app/jobs/destroy_downloads_job.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# Destroy a download, currently launched from the DownloadsController after a link is clicked.
+class DestroyDownloadsJob < ApplicationJob
+  queue_as :default
+
+  def perform(download)
+    download.destroy
```

Comment:
Will this also destroy from AWS?

Reply:
Yes calling destroy launches a background job that removes from AWS. this is internal to activestorage.